File: Question/Question 21-30/28. Implement strStr().java
class Solution {
    public int strStr(String haystack, String needle) {
        if(needle == null || haystack == null || haystack.length() < needle.length()){
            return -1;
        }
        if("".equals(needle) || haystack.startsWith(needle)){
            return 0;
        }
        String[] strArr = haystack.split(needle);
        return  (strArr.length == 1 && (strArr[0].length() < haystack.length())) || strArr.length > 1 ? strArr[0].length() : -1;
    }
}

File: Question/Question 21-30/28. Implement strStr().js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
let strStr = function(haystack, needle){
    haystack = haystack||"";
    if(needle === "" || haystack.startsWith(needle)){return 0;}
    return haystack.indexOf(needle);
}